src/core/core.interaction.js

```diff
@@ -42,35 +42,32 @@ function evaluateAllVisibleItems(chart, handler) {
 }
 
 /**
- * Helper function to check the items at the hovered index on the index scale
+ * Helper function to get items using binary search, when the data is sorted.
  * @param {Chart} chart - the chart
  * @param {string} axis - the axis mode. x|y|xy
  * @param {object} position - the point to be nearest to
  * @param {function} handler - the callback to execute for each visible item
  * @return whether all scales were of a suitable type
  */
-function evaluateItemsAtIndex(chart, axis, position, handler) {
+function binarySearchItems(chart, axis, position, handler) {
 	const metasets = chart._getSortedVisibleDatasetMetas();
-	const indices = [];
 	for (let i = 0, ilen = metasets.length; i < ilen; ++i) {
 		const metaset = metasets[i];
 		const iScale = metaset.controller._cachedMeta.iScale;
-		if (!iScale || axis !== iScale.axis || !iScale.getIndexForPixel) {
+		if (!iScale || axis !== iScale.axis || !metaset._sorted) {
 			return false;
 		}
-		const index = iScale.getIndexForPixel(position[axis]);
```

Comment:
This is O(1) whereas binary search is O(log n), so I expect it to be slower for time series charts

Reply:
Yeah, its about 2x faster. ~10 sec interaction in uPlot:
master:
![image](https://user-images.githubusercontent.com/27971921/72363504-a28d0c80-36fd-11ea-9c78-c297576a8dfe.png)

pr:
![image](https://user-images.githubusercontent.com/27971921/72363593-c6e8e900-36fd-11ea-8220-f30a8d4f1813.png)

Reply:
No call count there, so not very well comparable result.

Reply:
Different test (in uPlot bench console):
```
var start = Date.now();
for (var i = 0; i < 100000; i++) {
	var ix = Math.round(Math.random() * 55550);
	var point = chart._metasets[0].data[ix];
	var rect = chart.canvas.getBoundingClientRect();
	var event = {
		type: 'mousemove',
		target: chart.canvas,
		clientX: rect.left + point.x,
		clientY: rect.top + point.y,
		cancelable: true,
		bubbles: true,
		view: window
	};
	chart.getElementsAtEventForMode(event, 'nearest', {intersect: false, axis: 'x'});
}
console.log(Date.now() - start);
```
pr: 1676
master: 1016

so ~65% slower

Reply:
not that it matters for this cmp, but getBoundingClientRect() is usually expensive. caching it is generally advisable.

Reply:
@leeoniya its called for each event in Chart.js, so that's a valuable note.

Reply:
@benmccann it was O(1), but it had different behaviour. It just returned the nearest tick and then somehow used that index to get data. I don't think the old behaviour was right at all. 

Also, the implementation of `getIndexForPixel` in the time scale (only place it's actually defined) has a bug in it. It references `me.numIndices` instead of `me._numIndices`

Reply:
I don't understand what you think is wrong about the old behavior. In the time series case there's a 1:1 mapping between ticks and and data

Reply:
The thing that is wrong with it, is the assumption that ticks equal data indices. That should not be assumed. It holds true currently, unless the range is limited by `min` / `max`, or ticks source = 'labels' (or not 'data' / 'auto').

Example chart where it misbehaves: https://codepen.io/kurkle/pen/JjoBpyV?editors=0010

Reply:
I thought ticks matching data only happened when the time scale was in `series` mode

Reply:
@etimberg you are correct, only in series mode.